feat: Add GitHub event handler setup command

raffaele-oplabs opened the pull request:
Introduce a new command to create environment variables from GitHub event data. This command decodes the base64 encoded GitHub event and exports the relevant variables with a specified prefix, enhancing CI workflow integration.

<!--
Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md
-->

**Description**

<!--
A clear and concise description of the features you're adding in this pull request.
-->

This pull request introduces a new command for setting up GitHub event handlers by creating environment variables from a base64 encoded GitHub event. The key changes are as follows:

* [`orb/src/commands/github-event-handler-setup.yml`](diffhunk://#diff-d60e0073fb4efcaaddb1db3e701a2859244452de10adb8e8e3311abb17207d16R1-R25): Added a new command to decode a base64 encoded GitHub event and create environment variables with a specified prefix. This includes parameters for the environment variable prefix and the base64 encoded event, as well as steps to decode the event and generate environment variables.

**Tests**

<!--
Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.
-->

**Additional context**

<!--
Add any other context about the problem you're solving.
-->

**Metadata**

<!-- 
Include a link to any github issues that this may close in the following form:
- Fixes #[Link to Issue]
-->

raffaele-oplabs committed:
feat: Add GitHub event handler setup command

Introduce a new command to create environment variables from GitHub event data. This command decodes the base64 encoded GitHub event and exports the relevant variables with a specified prefix, enhancing CI workflow integration.

janjakubnanista commented on orb/src/commands/github-event-handler-setup.yml:
```diff
@@ -0,0 +1,25 @@
+          command: |
+            GITHUB_EVENT_JSON=$(echo $github_event_base64 | base64 --decode)
+            # Decode the base64 json
+            echo "export ${env_prefix}json_event='$GITHUB_EVENT_JSON'" >> $BASH_ENV
```
What about single quotes here? Bash interpolation always scared me, you never know

raffaele-oplabs replied:
Good point, this is not managed safely even if from trusted source.
I will fix this.

raffaele-oplabs replied:
Fixed and tested on this [pipeline](https://app.circleci.com/pipelines/github/ethereum-optimism/testing_actions/6243/workflows/14cc2744-3382-45ea-b0d0-ad41f6d901c5/jobs/6437) and [commit](https://github.com/ethereum-optimism/testing_actions/blob/679bd9c18dd325101359c6a8e81536f47020e26c/.circleci/config.yml#L211).
 
![Screenshot 2025-04-09 at 18 15 17](https://github.com/user-attachments/assets/52bd3357-535d-4af8-a31e-89f60e8640a7)

very good catch. 

Can you check if that works for you?

jelias2 commented:
Where does this come into play in the event handler workflow?

raffaele-oplabs committed:
refactor: Enhance GitHub event handler setup command

Improve the command for creating environment variables from GitHub event data by adding JSON validation and safe escaping for variable exports. This update ensures better error handling and security when processing the base64 encoded input.
